app/services/suggester/users/sidebar.rb

```diff
@@ -4,33 +4,39 @@ class Sidebar
       def initialize(user, given_tag)
         @user = user
         @given_tag = given_tag
+        @minimum_reaction_count = Rails.env.production? ? 25 : 0
       end
 
       def suggest
-        Rails.cache.fetch(generate_cache_name, expires_in: 120.hours) do
-          reaction_count = Rails.env.production? ? 25 : 0
-          user_ids = Article.published.tagged_with([given_tag], any: true).
-            where("public_reactions_count > ?", reaction_count).
-            where("published_at > ?", 4.months.ago).
-            where("user_id != ?", user.id).
-            where.not(user_id: user.following_by_type("User")).
-            pluck(:user_id)
-          group_one = User.select(:id, :name, :username, :profile_image, :summary).
-            where(id: user_ids).
-            order("reputation_modifier DESC").limit(20).to_a
-          group_two = User.select(:id, :name, :username, :profile_image, :summary).
-            where(id: user_ids).
-            order(Arel.sql("RANDOM()")).limit(20).to_a
-          (group_one + group_two).uniq
+        suggested_user_ids = Rails.cache.fetch(generate_cache_name, expires_in: 120.hours) do
+          (reputable_user_ids + random_user_ids).uniq
         end
+        User.select(:id, :name, :username, :profile_image, :summary).where(id: suggested_user_ids).to_a
```

Comment:
is there a reason to make this array instead of keeping it an active record relation?

Reply:
I was keeping to identical to the original spec but you are right, it doesn't have to be an array. Updated